unity-client/Assets/UnityGLTF/Scripts/GLTFSceneImporter.cs

```diff
@@ -1111,23 +1111,50 @@ void ProcessCurves(Transform root, GameObject[] nodes, AnimationClip clip, GLTFA
 
             for (var ci = 0; ci < channelCount; ++ci)
             {
+                var optimizedKeyframes = OptimizeKeyFrames(keyframes[ci]);
                 // copy all key frames data to animation curve and add it to the clip
-                AnimationCurve curve = new AnimationCurve(keyframes[ci]);
+                AnimationCurve curve = new AnimationCurve(optimizedKeyframes);
 
                 // For cubic spline interpolation, the inTangents and outTangents are already explicitly defined.
                 // For the rest, set them appropriately.
                 if (mode != InterpolationType.CUBICSPLINE)
                 {
-                    for (var i = 0; i < keyframes[ci].Length; i++)
+                    for (var i = 0; i < optimizedKeyframes.Length; i++)
                     {
-                        SetTangentMode(curve, keyframes[ci], i, mode);
+                        SetTangentMode(curve, optimizedKeyframes, i, mode);
                     }
                 }
 
                 clip.SetCurve(relativePath, curveType, propertyNames[ci], curve);
             }
         }
 
+        private Keyframe[] OptimizeKeyFrames(Keyframe[] rawKeyframes)
+        {
+            List<Keyframe> result = new List<Keyframe>(2);
+
+            Keyframe lastKf = rawKeyframes[0];
+            result.Add(lastKf);
+
+            for (int i = 1; i < rawKeyframes.Length; i++)
+            {
+                float valueDelta = Mathf.Abs(rawKeyframes[i].value - lastKf.value);
+                float deviation = valueDelta;
+
+                if (deviation > 0.001f)
+                {
+                    result.Add(rawKeyframes[i]);
+                }
+
+                lastKf = rawKeyframes[i];
```

Comment:
Why isn't this part of the previous IF?

Reply:
Hi @menduz !

This is done in order to don't mess with the original data and add the correct ending keyframe on the `if (result.Count == 1)` section of this method. As keyframes have a `time` field, if I don't do that we'd end with `time == 0` for the keyframes in both sides of the curve. 

In any case I am still working on some tests that should confirm there isn't something that break the curves with this logic, because we could end with far away keyframes that have differing tangents and don't match the original animation closely. This is supposed to be a very lazy optimization, not a curve matching keyframe reduction that would require far more involved code.